airflow/providers/amazon/aws/sensors/s3.py

```diff
@@ -93,6 +96,8 @@ def __init__(
         self.check_fn = check_fn
         self.aws_conn_id = aws_conn_id
         self.verify = verify
+        self.deferrable = deferrable
+        self.should_check_fn = True if check_fn else False
```

Comment:
I don't see the benefit of pre-computing this. I'd inline this code where it's used (in the `defer` call)

Reply:
`check_fn` is python callable and it is not serializable as discussed here: https://github.com/apache/airflow/pull/31018#discussion_r1210612988

Reply:
what I mean is that instead of computing it here and storing it in a class field, you could just do that `True if check_fn else False` in the only place where you use it